smac/optimizer/smbo.py

```diff
@@ -13,19 +13,100 @@
 from smac.optimizer.random_configuration_chooser import ChooserNoCoolDown, RandomConfigurationChooser
 from smac.optimizer.ei_optimization import AcquisitionFunctionMaximizer
 from smac.optimizer.epm_configuration_chooser import EPMChooser
-from smac.runhistory.runhistory import RunHistory
+from smac.runhistory.runhistory import RunHistory, RunInfo
 from smac.runhistory.runhistory2epm import AbstractRunHistory2EPM
 from smac.scenario.scenario import Scenario
 from smac.stats.stats import Stats
-from smac.tae.execute_ta_run import FirstRunCrashedException
+from smac.utils.constants import MAXINT
+from smac.tae.execute_ta_run import (
+    FirstRunCrashedException,
+    StatusType,
+    ExecuteTARun,
+    CappedRunException,
+    BudgetExhaustedException
+)
 from smac.utils.io.traj_logging import TrajLogger
 from smac.utils.validate import Validator
 
+from logging import Logger
+
+
 __author__ = "Aaron Klein, Marius Lindauer, Matthias Feurer"
 __copyright__ = "Copyright 2015, ML4AAD"
 __license__ = "3-clause BSD"
 
 
+def eval_challenger(
```

Comment:
I don't see why this function lives in `smbo.py`. Seems more like a tae function.

Reply:
Definitely, moving it out.